### ISSUE TITLE
Feedback on initial SNODAS application

### ISSUE THREAD

**Issue by smalers** [OpenWaterFoundation/owf-app-snodas-ng#3]
Here is feedback on the initial application.  I'm putting in one issue but issues can be split as appropriate. Some of these are cosmetic and addressing now will avoid having to re-comment on later.  Some of these changes will also demonstrate the value of Angular and updates to the app, rather than just making it appear exactly as before.

1. Some of the fonts on the right seem a bit "skinny".  The font used for the main menu is nice so I'd make them consistent.
2. Zoom is too drastic.  I recommend using fractional zoom increments similar to InfoMapper, but try it out to make sure it works well.
3. It is a bit distracting to see background text show through the data popup.  Maybe make it more opaque.
4. Would be nice to make the graph buttons fit on the page without scrolling for a normal-sized display (23"?).  Get rid of extra padding where possible.  Can probably get ***Current SNODAS Date*** to fit on one line by using smaller font.
5. Should a progress indicator be shown after a new date is selected?  It may be slower for some users.
6. Maybe make the animation more clearly a separate control, like adding a border around it?  Maybe move it to the bottom of the right panel since it is a secondary feature and picking date and basin are the primary interactions?  Or, maybe make tabs in the right panel with "View Data" and "Animate" (as long as the two use cases are separate).
7. Is the animation period selector constrained to the limits of the data?  It needs to be - otherwise people could pick bad dates.
8. Maybe show selected basin in a different color than mouseover highlight?  Yellow is often used for select/highlight but does not work as well on light-colored background.
9. Image viewer window works well.  However, it is not resizable so can the resize handle on the dialog be omitted?  Or, allow resize and make it work somehow.  Controlling the dialog behavior could be properties in the library code.  I'm interested in reviewing the properties only because I have done a lot of that type of naming over the years.  Such properties are also important for the code API and I am interested in whether `README.md` is being used to document for developers.
10. Nice that multiple graphs can be shown at the same time.  However, I could not open more than one SWE graph image at the same time.  If viewing multiple graphs, being able to resize would be nice.
11. README should describe which basins have test data.
12. **Need a way of pointing the application to the full dataset hosted on State's server.  This should maybe be the default?  Then use a URL query parameter or other way to switch to developer mode?  I need to be able to test full functionality with data that are current and available on [http://snodas.cdss.state.co.us/app/index.html](http://snodas.cdss.state.co.us/app/index.html)**
13. The graph buttons need to all show the images and be enabled.  Any initial work to implement interactive graphs can be a second update to the application once the basic functionality is in place.
14. Maybe add a label above buttons with ***SNODAS Data Graphs;*** and then remove the word "SNODAS" from all buttons in order to shorting labels.
15. Centered name and ID are a bit distracting.  Maybe left-justify the labels so things don't move around so much?
16. Select basin by typing has some odd behaviors:
    1. Shows random words (from other browser searches?) - should only use words in the names.
    2. I tried typing "ft collins" but after the `ft` it showed a list and entry area went away.  Is the entry area behind the popup list?  Does a space cause something to get selected?
17. Change the documentation to Markdown and use showdown library (has this been done)?  I can edit the content more easily if that is changed.
18. Data menu should be enabled as another Markdown that describes links.
19. Going to other menus and then back to ***Map*** resets the map to original settings.  Can it remember its state from before?
20. Add favicon.  This will need to be configurable because if we can convince the State to use the new version, it should probably use their current logo (see TSTool documentation).  I may try to see if we can jointly brand the product in this case.
21. Maybe change the map ***Current Date**** to "Map Data Date***.  I think this is most relevant for the animation and in other cases would agree with the selected date.
22. Date selector should display dates in ISO 8601 format for consistency.  I try to use YYYY-MM-DD everywhere since it is computer friendly and matches data files.

**Comment by Nightsphere** [comment]
Here are my quick notes on each point from the previous feedback comment. This is also for my benefit to see what I've done at this point and what I have left:

1. Fixed
2. Changed to InfoMapper way of zooming.
3. Opacity changed from 0.8 to 0.95. Any more will just make it fully opaque.
4. Put animation into a `mat-accordian` and `mat-expansion-panel` to save space.
5. Looking into it.
6. See 4.
7. It is constrained, but more robust solutions need to be implemented. For example, picking dates from the dropdown will cut off dates accordingly, but nothing is stopping a user from manually entering whatever date they want.
8. I'll ask Steve for more clarification.
9. Image dialogs are resizable now, but more will need to be looked into when taking image descriptions into account.
10. Multiple graphs can now be opened.
11. I'll ask Steve for more clarification.
12. Images are now being taken from the State server, so all graph buttons will work. Dev mode still needs to be looked into.
13. Fixed.
14. Fixed.
15. Fixed.
16. Looking into it.
17. Updated. The markdown files displayed under `About` and `Documentation` are now under `assets/docs` and named for each tab.
18. I'll ask Steve for more clarification.
19. Looking into it.
20. Updated, with the option to configure easily set up later.
21. Updated.
22. Fixed.

**Comment by smalers** [comment]
Here are my comments.

1. Fonts are good.
2. Zoom is good.
3. Opacity is good.
4. Animation layout at the bottom is good.   There still seems to be extra space around the buttons, which causes scrolling to be added.  Can spacing between buttons be tightened up without seeming crowded so that the animation label is more visible and less at risk of having to scroll?  It seems like single-line buttons have more buffer and can be trimmed down.  The "Select Date" and "Select Basin" choices could also perhaps be made not so high.
5. ...
6. See 4.
7. Add as a separate issue.
8. Add as a separate issue and deal with later.
9. Image viewer dialog seems to have extra space at the bottom.  Also, it is distorting the image.  I recommend that that the relative aspect of images always be retained.  In other words, don't stretch in either dimension without changing the other dimension.  Otherwise, the text is hard to read. For this application, the original image size should be retained because images have been created to produce readable information products.  The main use case is that users might make smaller to show multiple images.  Think of the images as a photograph, which should be resized without distortion.
10. OK.
11. Mainly I want developers to know what basins can be used for development.  And, there needs to be clear documentation on how to use full data (currently the default) and test data (need to enable somehow to allow new features to be added with data that might not be the same format as public data).
12. Seems good to select all basins.  However, the list of dates seems to be using local data.  Need to ensure that data are either all remote, or all local.
13. OK
14. Now that "Graphs" is in a label above the buttons, remove "Graph" from all the buttons to further streamline labels.  Then perhaps only the last button will require 2 lines and there will be less issues with vertical layout scrolling.
15. OK
16. Maybe add as a separate issue.
17. I will review and make some changes.
18. Can't the original Data menu contents be converted to a Markdown file with links, similar to other tabs?  It may make sense to put links in a table to make it easier to read.  I can edit once enabled.
19. Maybe add as a separate issue.
20. Seems to be using the old favicon.  See [this file](https://github.com/OpenCDSS/cdss-app-statemod-fortran-doc-user/blob/master/mkdocs-project/docs/images/DWR_icon.gif).
21. OK
22. I will review more later when animation is fully enabled.
    1. The label should be "Enter a date range".  Internally, common date formats should be parsed, including ISO 8601 and American dates.  The Java DateTime parse method did this and I assume that JavaScript libraries will handle.  However, for output that we have control over, display as ISO 8601 format. 

New feedback:

23. Need to change the style for disabled buttons.  They are a bit hard to read because too faded.
24. Animation buttons need tooltips

**Comment by Nightsphere** [comment]
The following list is what I've done with the remaining issues (using bullets as I'm skipping numbers that are completely done and markdown numbered lists wont let me skip):

* 2: Initial zoom was bumped up to `7.5` so Colorado takes up a bit more of the screen.
* 4: Button sizing and other sizing in the side-nav has been updated so scrolling is not added.
* 7: Added as Issue #4.
* 8: Added as Issue #5.
* 9: Working on this now.
* 11: I will add to the README
* 12: CORS reared its head on this, so I've made this into Issue #9.
* 14: Updated.
* 16: Added as Issue #6.
* 19: Added as Issue #7.
* 20: Updated.
* 22: This has been updated. Luckily, the Material Datepicker has the conversion built in. Whether a user inputs `1/1/2017`, `01/01/2017`, `2017-01-01`, or `2017-1-1`, it will be converted to `2017-01-01` and the Datepicker will obtain the correct date.
* 23: Added as Issue #8.
* 24: Updated. I have also disabled the animation buttons until the Submit button is clicked. Other than that, clicking the Submit button doesn't do anything else for the moment.
* 25: Updated.

**Comment by Nightsphere** [comment]
Update to number 9, the resizing of image dialogs. The image aspect ratio is now kept on a resize, and the common library version has been updated from `0.0.1-alpha.4` to `0.0.1-alpha.5`.

**Comment by Nightsphere** [comment]
All numbered issues/additions for the initial SNODAS conversion have either been completed, or opened up as their own separate issues. Closing this issue.
